src/celmodel/mesh.cpp

```diff
@@ -29,60 +32,48 @@ constexpr std::size_t VertexAttributeFormatSizes[static_cast<std::size_t>(Vertex
      4,  // UByte4
 };
 
-} // end unnamed namespace
 
-VertexDescription::VertexDescription(unsigned int _stride,
-                                           unsigned int _nAttributes,
-                                           VertexAttribute* _attributes) :
-    stride(_stride),
-    nAttributes(_nAttributes),
-    attributes(nullptr)
+template<typename It>
+VertexDescription appendingAttributes(const VertexDescription& desc, It begin, It end)
 {
-    if (nAttributes != 0)
+    std::vector<VertexAttribute> allAttributes;
+    allAttributes.reserve(desc.attributes.size() + (end - begin));
```

Comment:
is (end-begin) guaranteed to return size_t?

Reply:
This will end up being the `difference_type` of the iterator. For `std::array` this will be `ptrdiff_t`.